File: frontend/src/app/dto/project.dto.ts
import {UserDto} from "./user.dto";
import {TaskDto} from "./task.dto";

export class ProjectDto {
  public projectId: number;
  public projectTitle: string;
  public startDate: string;
  public endDate: string;
  public priority: number;
  public userId: number;
  public userDtos: UserDto[];
  public taskDtos: TaskDto[];
  public totalNoOfTasks: number;
  public totalNoOfCompletedTasks: number;
}

File: backend/src/main/java/com/pm/dto/ProjectDto.java
package com.pm.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
public class ProjectDto {
    private long projectId;
    private String projectTitle;
    private LocalDate startDate;
    private LocalDate endDate;
    private int priority;
    private long userId;
    private List<UserDto> userDtos = new ArrayList<>();
    private List<TaskDto> taskDtos = new ArrayList<>();
    private int totalNoOfTasks;
    private int totalNoOfCompletedTasks;
}
